chsdi/views/catalog.py

```diff
@@ -1,14 +1,86 @@
 # -*- coding: utf-8 -*-
 
 from pyramid.view import view_config
-from pyramid.httpexceptions import HTTPNotFound
+import networkx as nx
+from itertools import chain
+from pyramid.httpexceptions import HTTPInternalServerError, HTTPNotFound
 
 from chsdi.models.bod import Catalog
 from chsdi.lib.validation import MapNameValidation
 from chsdi.lib.sqlalchemy_customs import remove_accents
 from chsdi.lib.filters import filter_by_geodata_staging
 
 
+# https://github.com/networkx/networkx/blob/master/networkx/readwrite/json_graph/tree.py#L16
```

Comment:
Have you add a look at add_children function? It is for sorting the children according to orderKey. I don't think I can come up with something generic enough. Note that it is the last function we use to prepare the json.